packages/openai/src/openai-provider.ts

```diff
@@ -249,3 +258,12 @@ export function createOpenAI(
 Default OpenAI provider instance.
  */
 export const openai = createOpenAI();
+
+declare module '@ai-sdk/provider-utils' {
+  interface RegisterProviderOptions {
+    openai?:
+      | OpenAIResponsesProviderOptions
+      | OpenAIChatProviderOptions
+      | OpenAICompletionProviderOptions;
```

Comment:
We think this might not be an improvement, the number of `*ProviderOptions` will grow. We think we want to invest into the explicit `import ...` and `... satisfies ...` approach for now. We do agree eventually we should go down the path of your PR eventually, we are just not ready yet.